### PR TITLE
Add basic-tutorial-3 from the GStreamer tutorials

### PR DESCRIPTION
**Pull request by lovebug356**
https://cgit.freedesktop.org/gstreamer/gst-docs/tree/examples/tutorials/basic-tutorial-3.c

### CODE REVIEW

**Comment by sdroege** [comment]
Thanks a lot, I'll review later :)

File: tutorials/src/bin/basic-tutorial-3.rs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        let convert = &convert_clone;
+
+        println!("Received new pad {} from {}",
+                 src_pad.get_name(),
+                 pipeline.get_name());
+
+        let sink_pad = convert.get_static_pad("sink")
+            .expect("  Failed to get static sink pad from convert");
+        if sink_pad.is_linked() {
+            println!("  We are already linked. Ignoring.");
```
Comment (sdroege) [inline tutorials/src/bin/basic-tutorial-3.rs]: Why the spaces at the front of the strings?

File: tutorials/src/bin/basic-tutorial-3.rs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    let pipeline = gst::Pipeline::new("test-pipeline");
+
+    // Build the pipeline Note that we are NOT linking the source at this
+    // point. We will do it later.
+    pipeline.add_many(&[&source, &convert, &sink]).unwrap();
+    convert.link(&sink).expect("Elements could not be linked.");
+
+    // Set the URI to play
+    let uri = "https://www.freedesktop.org/software/gstreamer-sdk/data/media/sintel_trailer-480p.webm";
+    source.set_property_from_str("uri", uri);
```
Comment (sdroege) [inline tutorials/src/bin/basic-tutorial-3.rs]: `source.set_property("uri", &uri)` seems nicer

Comment (lovebug356) [reply tutorials/src/bin/basic-tutorial-3.rs]: fixed!

File: tutorials/src/bin/basic-tutorial-3.rs (hunk 1 above)
Comment (lovebug356) [reply tutorials/src/bin/basic-tutorial-3.rs]: I was just copying the strings from the C examples, they also have some spaces in front. Removed them now.

Comment (sdroege) [reply tutorials/src/bin/basic-tutorial-3.rs]: Want to send a patch for the C ones too?

**Pull request closed without merging**